src/object.d

```diff
@@ -111,18 +111,34 @@ int __cmp(T)(scope const T[] lhs, scope const T[] rhs) @trusted
     }
 }
 
-// Compare class and interface objects for ordering.
-private int __cmp(Obj)(Obj lhs, Obj rhs)
-if (is(Obj : Object))
+// Compare class objects for ordering.
+int __cmp(C1, C2)(C1 lhs, C2 rhs)
```

Comment:
Can this be private and the compiler bypasses the protection? Or is there a use case for calling this directly?